dask/array/tests/test_array_core.py

```diff
@@ -217,6 +217,30 @@ def test_Array_computation():
     assert float(a[0, 0]) == 1
 
 
+@pytest.mark.skipif(LooseVersion(np.__version__) < '1.14.0',
+                    reason="NumPy doesn't have `np.linalg._umath_linalg` yet")
+def test_Array_numpy_gufunc_call__array_ufunc__01():
+    x = da.random.normal(size=(3, 10, 10), chunks=(2, 10, 10))
+    nx = x.compute()
+    ny = np.linalg._umath_linalg.inv(nx)
```

Comment:
Can you use the public `np.linalg.inv` with `apply_gufunc`? It's not a good idea to rely on functions in private modules, because those could change in any release without warning.

I guess we could stick with this (since you document these methods in the docs, and numpy doesn't expose any gufuncs in its public API currently), but it makes me a little nervous. Maybe it would be better to make the `skipif` condition `hasattr(hasattr(np.linalg, '_umath_linalg'), 'inv'`)`.

Reply:
Will do tomorrow
- [ ] Make _umath_linalg tests optional

Reply:
The problem seems, that the private functions exist already, but do not receive optional `kwargs`

Using `numpy=1.11`:
```
____________________________________ test_Array_numpy_gufunc_call__array_ufunc__01 _____________________________________

    @pytest.mark.skipif(hasattr(hasattr(np.linalg, '_umath_linalg'), 'inv'),
                        reason="NumPy doesn't have `np.linalg._umath_linalg.inv`")
    def test_Array_numpy_gufunc_call__array_ufunc__01():
        x = da.random.normal(size=(3, 10, 10), chunks=(2, 10, 10))
        nx = x.compute()
        ny = np.linalg._umath_linalg.inv(nx)
>       y = np.linalg._umath_linalg.inv(x, output_dtypes=float)
E       TypeError: 'output_dtypes' is an invalid keyword to ufunc 'inv'

dask/array/tests/test_array_core.py:226: TypeError
=============================================== 1 failed in 0.76 seconds ===============================================
```